src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -286,8 +291,10 @@ protected static async Task<bool> MatchesAsync(RecordValue currentRecord, Record
                 return false;
             }
 
+            cancellationToken.ThrowIfCancellationRequested();
             await foreach (var baseRecordField in baseRecord.GetFieldsAsync(cancellationToken).ConfigureAwait(false))
             {
+                cancellationToken.ThrowIfCancellationRequested();

```

Comment:
this one seems redundant with 295 (the one passed into the for loop)

Reply:
we're in an await foreach loop and not running in the same Task context so we need to check the CT again and in each loop iteration too, no?